receiver/azuremonitorreceiver/scraper.go

```diff
@@ -319,6 +338,64 @@ func (s *azureScraper) getResources(ctx context.Context, subscriptionID string)
 	s.subscriptions[subscriptionID].resourcesUpdated = time.Now()
 }
 
+// hackResources is a workaround specially done for the storageAccount metrics.
+// Every StorageAccount resources have some implicit sub resources (/blobServices/default, fileServices/default, etc...) that are not returned by the API.
+// We need to add them manually to the resources and resourceTypes map.
+// Note that we do that hack only if the user has asked the sub resource types explicitly in the services config.
+// Example:
+// For each resource with id .../Microsoft.Storage/storageAccount/myResource of type Microsoft.Storage/storageAccount,
+// It will create a fake resource with id .../Microsoft.Storage/storageAccount/myResource/blobServices/default of type Microsoft.Storage/storageAccounts/blobServices.
+// TODO: duplicate
+func (s *azureScraper) hackResources(resources []*armresources.GenericResourceExpanded) []*armresources.GenericResourceExpanded {
+	var hackedResources []*armresources.GenericResourceExpanded
```

Comment:
I have a feeling the verb `hacked` may scare some users that are just glancing through the project, could you use a different name here?